File: Scripts/laptop/call-an-object-method/c-sharp/call-an-object-method.cs
// Static
//MyClass.Method(someParameter);

// Instance
//myInstance.Method(someParameter);

using System;

class MyClass{

  // instance variable
  volatile int variable;	// Note: instance variables are usually "private"

  /**
  * The constructor
  */
  public MyClass(){
    // creates a new instance
  }

  /**
  * A method
  */
  public void someMethod(int value, int i){
   this.variable = value + i;
 }

	private static volatile int r = 1;

	public static int executeTask(int i) {

		MyClass obj = new MyClass();
		obj.someMethod(1, i);
		return 1 ;
	}

public static void Main(String[] args){

	for (int i = 0; i < 2000000000; ++i) {
		r = executeTask(i);
	}
}

}

File: Scripts/server/call-an-object-method/go/call-an-object-method.go
package main

type MyGoClass struct {
    variable  int // "instance variables"
}

func  someMethod(b *MyGoClass) {
	obj_2 :=&MyGoClass{variable: b.variable}
	_ = obj_2
}

func executeTask(i int) int {

	obj := &MyGoClass{variable: 1000}
	someMethod(obj)
	return i + 1;
}

func main(){
r := int(1)    

for i := 0; i < 2000000000; i++ {
	r = executeTask(i + r)
}
}
